### DISCUSSION TITLE
Add a Multimeter component

### DISCUSSION THREAD

**Discussion by StruggleDuck** [FakeFishGames/Barotrauma#7159] [Ideas and suggestions]
Very simple addition that I am quite honestly surprised doesn't already exist. the basic concept of the multimeter component is that it would be able to read and output certain information to do with power.

![CheckPower](https://user-images.githubusercontent.com/92034548/140082071-79c1a2fd-9a5b-4a96-a82e-10395a187d21.jpg)
As you can see in the image, the component would be able to be essentially be placed in-between powered connections, and then be able to output certain information about how much power is being being drawn by the components being powered, as well as how much power and voltage is going through the component. It could also be capable of outputting a signal if certain conditions are met in regards to how much power is going through, if power is going through, etc. 

I believe that this would be great for sub builders, because it would allow for the development of more complex power grids/systems. It would also be quite useful for testing your subs, and seeing where possible issues may arise, and what systems are using the most power. Fingers crossed that we see this become an addition to the game at some point in the future.

**Comment by StruggleDuck** [comment]
I just wanted to write down a couple of ideas that I had come up with if I had access to a component like this

- system that automatically temporarily shuts down certain systems to not overload the power grid
- system that automatically reduces the subs speed to make sure that the engine does not draw too much power
- system that calculates and then displays the amount of power each section of the ship is using
- system that calculates how long a battery will last at its current consumption rate
- system that alerts crew members if critical systems are no longer receiving power (i.e. oxygen generator, engine, weapon systems)
- many others that I am forgetting at the moment.

**Reply by mLuby** [reply]
How about ["Multimeter"](https://en.wikipedia.org/wiki/Multimeter) Component instead of "Check Power" Component?

**Reply by StruggleDuck** [reply]
that would probably be a much more suitable name, at the time of making this discussion I couldn't think of a better name lol

**Reply by MaaiG** [reply]
> that would probably be a much more suitable name...

You can change the title at any time! ))
![Image 002_17-39](https://user-images.githubusercontent.com/93348499/145591998-26a81321-3f1c-406a-8d2c-789053e94f54.png)

**Reply by StruggleDuck** [reply]
fixed, thanks

**Comment by MaaiG** [comment]
Yea, but OUT on the right side ))

So maybe you can tell me what it’s for and how to use it? I don’t get how it work...
![Image 001_23-52 (1)](https://user-images.githubusercontent.com/93348499/144921549-9685ab2b-6e6c-4be7-87c9-a67f0e26958d.png)

**Reply by gerakis** [reply]
It combines inputs to outputs. So any wire attached to the same pin will spread its signal to the rest of the wires on that pin.

![how to use signal in junction boxes](https://user-images.githubusercontent.com/16128947/145128184-42506e34-24f3-4a7f-8831-6a97f81d95ec.png)

**Reply by MaaiG** [reply]
> It combines inputs to outputs. 

Thanks! I never would have guessed...

**Comment by gerakis** [comment]
They could add all this to relay components. The ability to read the load on one pin, and another pin for power supplied would be hugely useful. You can already get this from the reactor, but not from relays.

**Comment by gerakis** [comment]
This right here is what I had in mind. Just add the same outputs to relays as the reactor has. You could probably copy paste the code. The relays already keep track of their load and power so all you have to do is add that to a pair of pins.

![improved relay](https://user-images.githubusercontent.com/16128947/145127557-f2fa33c9-fc74-430c-8b9f-7eb22d70609c.png)

**Reply by StruggleDuck** [reply]
The other option would be adding the pins to junction boxes, as seen in this pull request: https://github.com/Regalis11/Barotrauma/pull/7303

**Comment by MaaiG** [comment]
Are the developers considering these proposals?
